### ISSUE TITLE
File modified with excel not synchronized

### ISSUE THREAD

**Issue by JulienDevillers** [nextcloud/android#3215]
### Actual behaviour
- A file already on the server, opened with excel android, modified and saved is not synchronized

### Expected behaviour
- It should be automatically synchronized
 
### Steps to reproduce
1. create a file on pc, it is synchronized with the server
2. open it : Nextcloud / navigate to file / open / choose excel as editor
3. modify the file
4. press "back", in excel top bar it is written as "saved"
5. wait and the **file is never synchronized**
**or**  ( see the step's numbers, That's important ) 
5. in nextcloud file browser, click "synchronize" menu of the excel file. **It is correctly synchronized**
**or**
2. in nextcloud file browser, click "set available offline" (sorry I don't kown the exact english translation)  menu of the excel file. 
3. open It
4. modify the file
5. press "back", in excel top bar it is written as "saved"
6. wait and the **file is synchronized**

### Environment data
Android version: 8.0.0

Device model: Galaxy S9

Stock or customized system: stock

Nextcloud app version: 3.3.1

Nextcloud server version:  14.0.3

#### Web server error log
```  N/A
Insert your webserver log here
``` N/A

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by JulienDevillers** [comment]
Hello to the bot.  ;)

Yes, looks a lot like #337, should it be considered as a regular behaviour?

For me, no. If I modify a text file with the internal editor, it is automatically sync. If I open a file with an external app, it is not sync. IMHO, the behaviour should be the same for both editor external or internal.

**Comment by AndyScherzinger** [comment]
@JulienDevillers what you described it the actual behavior of the client as it has been designed. We are currently working on a change of this behavior, see #2886 which should take care syncing back any changed file to the server when changed on the client.

Please check if #2886 would actually change the behavior as expected by you :)

**Comment by JulienDevillers** [comment]
Yes, looks nice except that I would like to decide myself wether sync only on wifi or not.

I need one highlight:
If you sync all downloaded files, will there be a way to say "I don't want this file to sync anymore" ie "undownload the file", ie remove the local copy, without removing it from the server.

Use cases : 
I have a big file, downloaded it.

Use case 1 : I need to save space on my phone. I want to remove it localy only.

Use case 2 : It is updated too often on the server, I don't want to have it in sync (or manual sync only).
For this use case, as I understand,the future feature, won't handle this. A workaround would be to be able to save it in a directory other than nextcloud's sync dirs and then remove it localy only.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky is this issue covered by your latest improvements to the document provider implementation as well as the keep downloaded files in sync change?

**Comment by tobiasKaminsky** [comment]
Depends if Excel is using the doc storage api.
@JulienDevillers please test it with upcoming 3.5.0 version.

**Comment by jakobroehrl** [comment]
@AndyScherzinger @tobiasKaminsky 
It's not working for me with the new version 3.5. 
After saving the file there is no automatic upload. 

What does "document provider implementation" and "doc storage api" exactly mean?

**Comment by JulienDevillers** [comment]
OK, waiting for 3.5.0
    Le mercredi 30 janvier 2019 à 12:22:17 UTC+1, Tobias Kaminsky <notifications@github.com> a écrit :  
 
 
Depends if Excel is using the doc storage api.
@JulienDevillers please test it with upcoming 3.5.0 version.

**Comment by jakobroehrl** [comment]
I tested again. The auto upload after editing a file does not work. 
@tobiasKaminsky: Is it possible to trigger the sync method after a file was changed?

**Comment by tobiasKaminsky** [comment]
Auto Upload is uploading new images/files if you have configured it.

Then there is additionally a new service that tries to update all downloaded files every 15min on wifi and this should then detect the changes.
I'll have a look.

**Comment by jakobroehrl** [comment]
> Auto Upload is uploading new images/files if you have configured it.
Is this the right way to configure it?
Menu -> automatic upload ->3 dots -> create own folder? 
Clicking on local folder kills the app.
 
> Then there is additionally a new service that tries to update all downloaded files every 15min on wifi and this should then detect the changes.
> I'll have a look.
That's a very nice feature, thank you! I would appreciate it very much. :)
Thanks for your great work!

**Comment by tobiasKaminsky** [comment]
> > Auto Upload is uploading new images/files if you have configured it.
> 
> Is this the right way to configure it?
> Menu -> automatic upload ->3 dots -> create own folder?
> Clicking on local folder kills the app.

You should also see existing folders, so you do not have to create your own folder.
Thank you for reporting the crash, I'll added a fix.



> That's a very nice feature, thank you! I would appreciate it very much. :)

Glad that you like it, please report back any problems you might encounter

**Comment by tobiasKaminsky** [comment]
A new idea is following:
- open excel file from NC
- select MS Excel (this does not use doc provider :/)
- edit & save file
- click back -> go to NC
- now NC knows that the file was edited and can sync it

Only downside on this approach is, that one always have to open it from within our app.
So we cannot open any excel file directly from Excel app…

**Comment by derekblankmccoy** [comment]
> Then there is additionally a new service that tries to update all downloaded files every 15min on wifi and this should then detect the changes.
> I'll have a look.

Where is this feature?

**Comment by jakobroehrl** [comment]
@tobiasKaminsky: Is this already implemented?

**Comment by af7567** [comment]
I am having this problem with Nextcloud app 3.14.1. I have recently started using Collabora on Android to edit a spreadsheet from my Nextcloud server. I open the ods file from within the Nextcloud app and it automatically opens in Collabora. After making changes and saving them then going back to the Nextcloud app the file isn't synced with the server no long how long I wait.

If I tap the file again in the Nextcloud app then it will upload the local version to the server (and also open it in Collabora again)

**Comment by AndyScherzinger** [comment]
@af7567 please open a new issue for that. Other than that this seems impossible since you are editing via collabora within the app so you actually edit on the server... It may well be that the client doesn't properly refresh right away though.

**Comment by af7567** [comment]
I have just looked more into what Collabora is, I didn't know there was an online version too. I assumed their Android app did all the editing on the device. I just tested again and added something to the ods file then saved and quit collabora and after 30 seconds the timestamp of the file on my PC did change to the current time - but without the new data. Only after I opened the file again on my mobile I see the nextcloud notification at the top to let me know it is busy syncing and then I see the new data on the PC too.
I will have to try another program for editing the spreadsheet to rule out weirdness with Collabora.

**Comment by AndyScherzinger** [comment]
I didn't even know there is a collabora Android app 🙄
